backend/pkg/api/data_access/notifications.go

```diff
@@ -67,7 +68,152 @@ const (
 )
 
 func (d *DataAccessService) GetNotificationOverview(ctx context.Context, userId uint64) (*t.NotificationOverviewData, error) {
-	return d.dummy.GetNotificationOverview(ctx, userId)
+	response := t.NotificationOverviewData{}
+	eg := errgroup.Group{}
+
+	// enabled channels
+	eg.Go(func() error {
+		var channels []struct {
+			Channel string `db:"channel"`
+			Active  bool   `db:"active"`
+		}
+
+		err := d.userReader.SelectContext(ctx, &channels, `SELECT channel, active FROM users_notification_channels WHERE user_id = $1`, userId)
+		if err != nil {
+			return err
+		}
+
+		for _, channel := range channels {
+			switch channel.Channel {
+			case "email":
+				response.IsEmailNotificationsEnabled = channel.Active
+			case "push":
+				response.IsPushNotificationsEnabled = channel.Active
+			}
+		}
+		return nil
+	})
+
+	getMostNotifiedGroups := func(historyTable, groupsTable string) ([3]string, error) {
+		query := goqu.Dialect("postgres").
+			From(goqu.T(historyTable).As("history")).
+			Select(
+				goqu.I("history.dashboard_id"),
+				goqu.I("history.group_id"),
+			).
+			Where(
+				goqu.Ex{"history.user_id": userId},
```

Comment:
this query should be limited to the last 30 days as the header of the corresponding view reads "Most Notifications Last 30d"